core/src/main/java/com/yandex/yoctodb/util/immutable/impl/BitSetIndexToIndexMultiMap.java

```diff
@@ -151,14 +155,41 @@ public String toString() {
     public Iterator<IntToIntArray> ascending(
             @NotNull
             final BitSet valueFilter) {
-        throw new UnsupportedOperationException();
+        final ArrayBitSet bs = LongArrayBitSet.zero(valueFilter.getSize());
+        return Stream.iterate(0, i -> i + 1)
+                .map(i -> getIntToIntArray(valueFilter, bs, i))
+                .limit(keysCount).iterator();
     }
 
     @NotNull
     @Override
     public Iterator<IntToIntArray> descending(
             @NotNull
             final BitSet valueFilter) {
-        throw new UnsupportedOperationException();
+
+        final ArrayBitSet bs = LongArrayBitSet.zero(valueFilter.getSize());
+        return Stream.iterate(keysCount - 1, i -> i - 1)
+                .map(i -> getIntToIntArray(valueFilter, bs, i))
+                .limit(keysCount).iterator();
+    }
+
+    @NotNull
+    private IntToIntArray getIntToIntArray(
+            @NotNull
+            final BitSet valueFilter,
+            final ArrayBitSet dest,
+            Integer i) {
```

Comment:
Do we really need boxed `Integer`s (and above)?

Reply:
Rewritten using IntStream with mapToObj, this way we get primitive int.